src/main/java/jarvis/Formatter.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    protected static boolean isValidCommand(int numOfArgs) {
+        return numOfArgs > 1;
+    }
+    protected static void markCommand(String[] userCommand) throws JarvisException {
+        int taskIndex = getTaskIndex(userCommand);
+        if (isValidIndex(taskIndex)) {
+            UserList.markTask(taskIndex, true);
+        } else {
+            throw new JarvisException();
+        }
+    }
+
+    protected static void unmarkCommand(String[] userCommand) throws JarvisException {
+        int taskIndex = getTaskIndex(userCommand);
+        if (isValidIndex(taskIndex)) {
+            UserList.unmarkTask(taskIndex);
+        } else {
+            throw new JarvisException();
+        }
+    }
+
+    protected static void todoCommand(String[] userCommand) throws JarvisException {
+        int numOfArgs = userCommand.length;
+        if (isValidCommand(numOfArgs)) {
+            String taskDescription = parseUserInput(userCommand, 1, numOfArgs);
+            Task newTask = new Task(taskDescription);
+            UserList.insertTask(newTask);
+        } else {
+           throw new JarvisException();
+        }
+    }
+
+    protected static void deadlineCommand(String[] userCommand) throws JarvisException {
+        int indexOfBy = indexOf(userCommand, "/by");
+        int numOfArgs = userCommand.length;
+        boolean isValidIndex = indexOfBy > 1;
+        boolean hasSufficientArgs = numOfArgs >= 4;
+
+        if (isValidIndex && hasSufficientArgs) {
+            String deadlineDescription = parseUserInput(userCommand, 1, indexOfBy);
+            String deadlineDate = parseUserInput(userCommand, indexOfBy + 1, numOfArgs);
+            Deadline newDeadline = new Deadline(deadlineDescription, deadlineDate);
+            UserList.insertTask(newDeadline);
+        } else {
+            throw new JarvisException();
+        }
+    }
+
+    protected static void eventCommand(String[] userCommand) throws JarvisException {
+        int indexOfAt = indexOf(userCommand, "/at");
+        int numOfArgs = userCommand.length;
+        boolean isValidIndex = indexOfAt > 1;
+        boolean hasSufficientArgs = numOfArgs >= 4;
+
+        if (isValidIndex && hasSufficientArgs) {
+            String eventDescription = parseUserInput(userCommand, 1, indexOfAt);
+            String eventDate = parseUserInput(userCommand, indexOfAt + 1, numOfArgs);
+            Event newEvent = new Event(eventDescription, eventDate);
+            UserList.insertTask(newEvent);
+        } else {
+            throw new JarvisException();
+        }
+    }
+
+    protected static void deleteCommand(String[] userCommand) throws JarvisException {
+        try {
+            Integer taskIndex = Integer.parseInt(userCommand[1]);
+            UserList.removeTask(taskIndex - 1);
+        } catch (NumberFormatException e){
+            throw new JarvisException();
+        } catch (IndexOutOfBoundsException e) {
+            throw new JarvisException();
+        }
```

Comment:
[Don't Repeat Yourself](https://nus-cs2113-ay2122s2.github.io/website/se-book-adapted/chapters/codeQuality.html#minimise-code-duplication). Consider combining this section to a single block.